### PR TITLE
Changes to make processed amount configurable

### PR DESCRIPTION
**Pull request by reshmabidikar**
- Changes to make processed amount configurable. 
- Some bug fixes (`paymentMethodId` was used while creating the `PluginPaymentTransactionInfoPlugin` instead of `kbTransactionId`).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,7 +72,21 @@ curl -v \
...
+
+```
+curl -v \
+-u'admin:password' \
+-H "X-Killbill-ApiKey: bob" \
+-H 'X-Killbill-ApiSecret: lazar' \
+-H "Content-Type: application/json" \
+-H 'X-Killbill-CreatedBy: stephane' \
+-X POST \
+--data-binary '{"METHODS":"purchasePayment", "AMOUNT":"20"}' \
```
Comment (pierre) [inline README.md:85]: Should be `10` per the sentence above ? 😄

File: src/main/java/org/killbill/billing/plugin/payment/PaymentTestPluginApi.java [hunk 2]
```diff
@@ -255,10 +277,13 @@ public PaymentTransactionInfoPlugin authorizePayment(final UUID kbAccountId,
                                                          final CallContext context) throws PaymentPluginApiException {
         final PaymentPluginStatus pluginStatus = handleState(properties);
         if (pluginStatus != null) {
+        	final BigDecimal configuredAmount = getAmount(properties);
+        	final BigDecimal amountToUse = configuredAmount != null ? configuredAmount : amount;
+        	
             final PluginPaymentTransactionInfoPlugin infoPlugin = new PluginPaymentTransactionInfoPlugin(kbPaymentId,
-                                                                                                         kbPaymentMethodId,
+            																							 kbTransactionId,
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/payment/PaymentTestPluginApi.java:284]: Nit: odd alignment? 🤔

File: src/main/java/org/killbill/billing/plugin/payment/dao/PaymentTestDao.java [hunk 3]
```diff
@@ -38,6 +39,7 @@
 import org.killbill.billing.plugin.payment.dao.gen.tables.records.TestpaymentPaymentMethodsRecord;
 import org.killbill.billing.plugin.payment.dao.gen.tables.records.TestpaymentResponsesRecord;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.google.common.collect.ImmutableList;
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/payment/dao/PaymentTestDao.java:42]: We're slowly moving away from Guava. You could use `List.of` here for instance instead. /cc @xsalefter

**Review by pierre — APPROVED** [review]
👏 for adding more tests. I don't think they are automatically run though.

You could add a `ci.yml` file similar to https://github.com/killbill/killbill-commons/blob/work-for-release-0.23.x/.github/workflows/ci.yml

Only the `travis` profile should be required (fast tests), although you might have to annotate the test methods with the right group (`@Test(groups="fast")`).

File: src/main/java/org/killbill/billing/plugin/payment/dao/PaymentTestDao.java (hunk 3 above)
Comment (xsalefter) [reply src/main/java/org/killbill/billing/plugin/payment/dao/PaymentTestDao.java:42]: ... to be precise, `Collections.emptyList()` as seems like you want to pass empty list.

**Review by sbrossie — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -2,9 +2,15 @@ name: ci
 
 on:
   push:
+    inputs:
+      jdk-matrix:
```
Comment (pierre) [inline .github/workflows/ci.yml:6]: This needs to be in the `with` block below (`inputs` is used by the shared script: the `with` arguments here become `inputs` on the other side).

**Comment by pierre** [comment]
🎉

**Pull request merged**